lib/active_model/serializer/reflection.rb

```diff
@@ -17,7 +17,28 @@ class Serializer
     #
     # So you can inspect reflections in your Adapters.
     #
-    Reflection = Struct.new(:name, :options) do
+    Reflection = Struct.new(:name, :options, :block) do
+      delegate :call, to: :reader
+
+      attr_reader :reader
+
+      def initialize(*)
+        super
+        @reader = self.class.build_reader(name, block)
+      end
+
+      def value(instance)
+        call(instance)
+      end
+
+      def self.build_reader(name, block)
```

Comment:
ref: https://github.com/rails-api/active_model_serializers/pull/1356/files#r46783854

I'm also not sure about 'value' as a name here... or even `instance` in place of `serializer_instance` or whatever.. just trying to get a feel for which names work better